api/src/main/java/org/eclipse/microprofile/concurrent/ManagedExecutorConfig.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.eclipse.microprofile.concurrent;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+/**
+ * <p>Configures an injected <code>ManagedExecutor</code> instance.</p>
+ *
+ * <p>Example usage:</p>
+ * <pre><code> &commat;Inject &commat;ManagedExecutorConfig(context=ThreadContext.CDI, maxAsync=5)
+ * ManagedExecutor executor;
+ * ...
+ * </code></pre>
+ */
+@Retention(RUNTIME)
+@Target(FIELD)
+public @interface ManagedExecutorConfig {
+    /**
+     * <p>Defines the set of thread context types to capture from the thread
+     * that creates a dependent stage (or that submits a task) and which to
+     * propagate to the thread where the action or task executes.</p>
+     *
+     * <p>The default set of thread context types is those required by the
+     * EE Concurrency spec, plus CDI.</p>
+     *
+     * <p>Constants for specifying some of the core context types are provided
+     * on {@link ThreadContext}. Other thread context types must be defined
+     * by the specification that defines the context type or by a related
+     * MicroProfile specification.</p>
+     *
+     * <p>Inclusion of a thread context type with prerequisites implies
+     * inclusion of the prerequisites, even if not explicitly specified.</p>
+     *
+     * <p>Thread context types which are not otherwise included in this set
+     * are removed from the thread of execution for the duration of the
+     * action or task.</p>
+     */
+    String[] context() default { ThreadContext.APPLICATION, ThreadContext.CDI, ThreadContext.SECURITY };
```

Comment:
Should ALL be the default?

Reply:
The prior default was chosen to match the EE Concurrency requirement for only a specific set of context types to be propagated by default, where
 @Inject @ManagedExecutorConfig ManagedExecutor executor1;
would create an instance with the same context propagation as
 @Inject ManagedExecutor executor2; 
which would also have the same context propagation as EE Concurrency's default instance,
 @Resource ManagedExecutorService executor3;
The idea was to allow a container that is both MP Concurrency and EE Concurrency to return the same instance from all of these.

That said, it would be possible to instead default to ALL and require the use of different instance than the EE Concurrency default instance if that would be more useful.

Reply:
Thinking about this some more, defaulting to ThreadContext.ALL would mean that if someone implemented a provider for transaction context that was capable of propagating transactions, then those would also propagate by default.  I don't think we want to go that far.  It seems better for performance and predictability that we default to the more limited set of core contexts.  The user can always specify ALL if they want it.

Reply:
```Thinking about this some more, defaulting to ThreadContext.ALL would mean that if someone implemented a provider for transaction context that was capable of propagating transactions, then those would also propagate by default.```

If an user provide a transaction context, surely the default one will be overriden. You won't need two of them.

Reply:
The 'someone' that I was referring wasn't intended to mean the end user.  I was referring to a container implementation of transactions (the transaction manager).  In the event that an end user did provide a conflicting implementation of one of already-provided context types, I think we discussed that would be considered an error path for now.  Allowing an override of context provided by someone else seems outright dangerous, and in the case of transactions would probably lead to data integrity issues.